### PR TITLE
MINIFI-184: Add Security Support

### PR DESCRIPTION
**Pull request by benqiu2016**

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: README.md [hunk 1]
```diff
@@ -243,6 +245,26 @@ Additionally, users can utilize the MiNiFi Toolkit Converter (version 0.0.1 - sc
                     Port: 10001
                     Host Name: localhost
 
+### Site2Site Security Configuration
+
+    in minifi.properties
```
Comment (apiri) [inline README.md]: We should stub these out as defaults in minifi.properties.  No values, but at least provide a quick way of having them be readily available.

File: cmake/FindOpenSSL.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#  OPENSSL_ROOT_DIR - Set this variable to the root installation of OpenSSL
```
Comment (benqiu2016) [inline cmake/FindOpenSSL.cmake]: if i do not provide that, build failed.

**Comment by benqiu2016** [comment]
@apiri new code commit to address your review comment, back out the IOS port.

File: cmake/FindOpenSSL.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#
+# Read-Only variables:
+#  OPENSSL_FOUND - system has the OpenSSL library
+#  OPENSSL_INCLUDE_DIR - the OpenSSL include directory
+#  OPENSSL_LIBRARIES - The libraries needed to use OpenSSL
+#  OPENSSL_VERSION - This is set to $major.$minor.$revision$path (eg. 0.9.8s)
+
+find_path(OPENSSL_INCLUDE_DIR
+    NAMES openssl/ssl.h
```
Comment (benqiu2016) [inline cmake/FindOpenSSL.cmake]: openssl is used pretty widely and very stable and they provide bug fix regularly to plug the security hole.

File: libminifi/include/FlowController.h [hunk 4]
```diff
@@ -90,6 +93,25 @@ class FlowController
 		}
 		return _flowController;
 	}
+	//! passphase for the private file callback
```
Comment (benqiu2016) [inline libminifi/include/FlowController.h]: All the central service like provenance and others are tied with flowController like current NiFi.
It make code more clear.

File: libminifi/src/FlowController.cpp [hunk 5]
```diff
@@ -75,22 +78,123 @@ FlowController::FlowController(std::string name)
...
 	if (!path)
 	{
         _logger->log_error("Could not locate path from provided configuration file name (%s).  Exiting.", full_path);
         exit(1);
     }
 
-    std::string pathString(path);
-    _configurationFileName = pathString;
-    _logger->log_info("FlowController NiFi Configuration file %s", pathString.c_str());
+	std::string pathString(path);
```
Comment (benqiu2016) [inline libminifi/src/FlowController.cpp]: Same as above

**Comment by benqiu2016** [comment]
@phrocker @apiri remove the FindOpenSSL cmake by setting the OPENSSL_ROOT_DIR. Thanks.

**Comment by apiri** [comment]
reviewing

**Comment by apiri** [comment]
Changes look good.  I saw there were a couple of inconsistencies with some of the docs and typos but corrected those and will incorporate in the merge.  Thanks for the fixes!

**Pull request closed without merging**